.github/workflows/build_test.yml

```diff
@@ -0,0 +1,25 @@
+name: build-and-run-tests
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+env:
+  CARGO_TERM_COLOR: always
+
+defaults:
+  run:
+    working-directory: compiler
+
+jobs:
+  build-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build
+        run: cargo build --verbose
+      - name: Initialize setup
+        run: cargo install cargo-insta
+      - name: Run tests
+        run: cargo insta test
```

Comment:
Why not use `just` commands here too?

Reply:
Since there was one `just` _recipe_ `just init`, installing `just` to run one recipe did not justify it. But we can add it. In future, we may have more _recipes_.

Reply:
Let's use `just`. I'll update.